src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.CinemaHall;
+import mate.academy.service.CinemaHallService;
+
+@Service
+public class CinemaHallServiceImpl implements CinemaHallService {
+    @Inject
+    private CinemaHallDao cinemaHallDao;
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return cinemaHallDao.add(cinemaHall);
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {
+        return Optional.ofNullable(cinemaHallDao.get(id).orElseThrow(() -> new
+                DataProcessingException("CinemaHall with id: "
+                + id
+                + " not found", null)));
```

Comment:
The `get` method uses `orElseThrow` to throw a `DataProcessingException` with a `null` cause. Consider providing a more meaningful exception message or handling the absence of a cinema hall more gracefully.